### DISCUSSION TITLE
How can it prevent Captcha resolver services such as 2Captcha?

### DISCUSSION THREAD

**Discussion by hckhanh** [prosopo/captcha#1541] [Q&A] [answered]
Just wondering how can it do that?

**Comment by forgetso** [comment, marked as answer]
Sorry, just seen this. 2captcha and other solving services are notoriously difficult to stop, which is why you can bypass most bot detection software on their website (Google (13s) / Cloudflare (15s) / DataDome (12s) / Arkose (28s) / Procaptcha (28s)). The other companies listed here are a LOT bigger than Prosopo.

We have had some luck detecting and blocking 2Captcha in the past:

<img width="678" height="708" alt="image" src="https://github.com/user-attachments/assets/5d7a8019-f546-48b2-830a-66541efb3235" />

And we plan on continuing to block them whenever possible. [Enterprise users](https://prosopo.io/pricing/) currently benefit from dynamically created [Access Control rules](https://prosopo.io/products/access-control/) which apply whenever suspicious behaviour is detected. We'll be rolling out a restricted free tier version of this at some point.

We also just released stricter controls over IPs matching through the process in our [last release](https://github.com/prosopo/captcha/releases/tag/v3.2.4). If you want to ensure that the IP of the person on your site is the same as the one we recorded, [you can pass it up in the verify call and we'll do some additional checks](https://docs.prosopo.io/en/basics/server-side-verification/#optional-ip-address).

I hope this clears things up! Please let me know if you have any questions.

**Reply by hckhanh** [reply]
thanks for response.
